### PR TITLE
chore(icons): deploy icons on nexus

### PR DESCRIPTION
**Pull request by rmannibucau**
**What is the problem this PR is trying to solve?**

Provide svg icons to maven based builds. Idea is to avoid to have to GET on github the icons when needed.

**What is the chosen solution to this problem?**

Add a webpack specific script which:
1. creates a jar/zip with all the icons
2. POST the zip to nexus

**TODO**

- [ ] Test Nexus deployment, it can need to be tuned (repository, etc...)
- [ ] Add a script to enable to publish, needs a `TALEND_RELEASE=true` for releases and nothing for snapshots (to deploy each time the same version suffixed with -SNAPSHOT)

**Please check if the PR fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/icons/webpack.maven.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+              });
+            });
+          } else {
+            console.log('[WARN] Artifact not ready to be uploaded');
+          }
+        });
+      }
+    }
+	]
+};
```
Comment (talend-glorieux) [inline packages/icons/webpack.maven.js]: File formatting is all wrong. Please make sure you comply with the common eslint rules.

File: packages/icons/webpack.maven.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	plugins: [
+    new CopyWebpackPlugin([{ from: 'src/svg' }]),
+    new ZipPlugin({
+      filename: `${mvnConfig.artifactId}-${mvnConfig.version}`,
+      extension: mvnConfig.packaging,
+      pathPrefix: 'icons',
+      include: [/\.svg$/],
+    }),
+    {
+      apply: function (compiler) {
```
Comment (talend-glorieux) [inline packages/icons/webpack.maven.js]: Could you separate that function into sub-function to make it easier to understand.

File: packages/icons/webpack.maven.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            };
+            const repository = process.env['NEXUS_REPOSITORY'] || (release ? 'releases' : 'snapshots');
+            const base = mvnConfig.repository + `/content/repositories/${repository}/${mvnConfig.groupId.replace(/\./g, '/')}/${mvnConfig.artifactId}/${mvnConfig.version}/${mvnConfig.artifactId}-${mvnConfig.version}.`;
+            const headers = {
+              'Cache-control': 'no-cache',
+              'Cache-store': 'no-store',
+              'Pragma': 'no-cache',
+              'Expires': '0',
+              'Accept-Encoding': 'gzip',
+              'User-Agent': 'Apache-Maven/3.5.0 (Java 1.8.0_131; Linux 4.10.0-30-generic)'
```
Comment (talend-glorieux) [inline packages/icons/webpack.maven.js]: Looks generated. Could you drop the JAVA & LINUX part?

File: packages/icons/package.json [hunk 4]
```diff
@@ -44,7 +45,10 @@
     "react": "15.5.4",
     "style-loader": "^0.13.1",
     "svgo": "0.7.1",
-    "webpack": "^1.14.0"
+    "webpack": "^1.14.0",
+    "zip-webpack-plugin": "2.0.0",
+    "copy-webpack-plugin": "4.0.1",
+    "request": "2.81.0"
```
Comment (talend-glorieux) [inline packages/icons/package.json]: Why not use fetch instead? It's already used on other packages.

**Comment by rmannibucau** [comment]
@talend-glorieux sorry for the delay, just updated the PR to address your comments.

**Comment by jsomsanith** [comment]
Replaced by #809

**Pull request closed without merging**